plugin-server/src/main/ingestion-queues/batch-processing/each-batch-buffer.ts

```diff
@@ -47,12 +46,6 @@ export async function eachBatchBuffer(
     if (consumerSleepMs > 0) {
         // pause the consumer for this partition until we can process all unprocessed messages from this batch
         await queue.bufferSleep(consumerSleepMs, batch.partition)
-
-        // we throw an error to prevent the non-processed message offsets from being committed
-        // from the kafkajs docs:
-        // > resolveOffset() is used to mark a message in the batch as processed.
-        // > In case of errors, the consumer will automatically commit the resolved offsets.
-        throw new DelayProcessing()
```

Comment:
The issue with throwing an error is it results in a lot of log spam and eventually `KafkaJSNumberOfRetriesExceeded` crashing everything (at least in the test I added). Why would `commitOffsetsIfNecessary()` not work here?

Reply:
whatever approach you can make work is fine by me

Reply:
key point is to not lose events

Reply:
It's unclear how this change affects us operationally. cc @yakkomajuri on this.

However based on this code change it seems to me that events that aren't processed immediately are then _never_ processed: see this:

```        if (delayUntilTimeToProcess < 0) {
            await eachMessageBuffer(message, resolveOffset, queue)
        } else {
            consumerSleepMs = Math.max(consumerSleepMs, delayUntilTimeToProcess)
        }```

Reply:
If retries are the problem - there might be a slightly hacky approach to throw a `KafkaJSNonRetriableError`